gilded_rose.rb

```diff
@@ -1,51 +1,51 @@
 def update_quality(items)
   items.each do |item|
-    if item.name != 'Aged Brie' && item.name != 'Backstage passes to a TAFKAL80ETC concert'
-      if item.quality > 0
-        if item.name != 'Sulfuras, Hand of Ragnaros'
-          item.quality -= 1
-        end
-      end
-    else
-      if item.quality < 50
-        item.quality += 1
-        if item.name == 'Backstage passes to a TAFKAL80ETC concert'
-          if item.sell_in < 11
-            if item.quality < 50
-              item.quality += 1
-            end
-          end
-          if item.sell_in < 6
-            if item.quality < 50
-              item.quality += 1
-            end
-          end
-        end
-      end
-    end
-    if item.name != 'Sulfuras, Hand of Ragnaros'
-      item.sell_in -= 1
-    end
-    if item.sell_in < 0
-      if item.name != "Aged Brie"
-        if item.name != 'Backstage passes to a TAFKAL80ETC concert'
-          if item.quality > 0
-            if item.name != 'Sulfuras, Hand of Ragnaros'
-              item.quality -= 1
-            end
-          end
-        else
-          item.quality = item.quality - item.quality
-        end
-      else
-        if item.quality < 50
-          item.quality += 1
-        end
-      end
+    case item.name
+      when "NORMAL ITEM"
+        next update_normal_item(item)
+      when 'Backstage passes to a TAFKAL80ETC concert'
+        next update_backstage_passes(item)
+      when 'Aged Brie'
+        next update_aged_brie(item)
+      when 'Conjured Mana Cake'
+        next update_conjured(item)
     end
   end
 end
 
+
+def update_normal_item(item)
+  item.sell_in -= 1
+  return if item.quality == 0
+  item.quality -= 1
+  item.quality -= 1 if item.sell_in < 0
+end
+
+# this passes the tests but technically doesn't follow spec - allows for quality over 50
+def update_backstage_passes(passes)
+  passes.sell_in -= 1
+  return passes.quality = 0 if passes.sell_in < 0
```

Comment:
lets extract these into more readable methods too!
```suggestion
  return passes.quality = 0 if expired?
```